File: app/src/data/user-role.ts
export enum UserRole {
    Editor = 0,
    Publisher = 1,
    Admin = 2,
}

export const UserRoleLabels: Record<UserRole, string> = {
    [UserRole.Admin]: 'Fachbereichs-Administrator:in',
    [UserRole.Publisher]: 'Veröffentlicher:in',
    [UserRole.Editor]: 'Bearbeiter:in',
};

export const UserRoleIdentifier: Record<UserRole, string> = {
    [UserRole.Admin]: 'Admin',
    [UserRole.Publisher]: 'Publisher',
    [UserRole.Editor]: 'Editor',
};

File: src/main/java/de/aivot/GoverBackend/enums/UserRole.java
package de.aivot.GoverBackend.enums;

import com.fasterxml.jackson.annotation.JsonValue;
import de.aivot.GoverBackend.lib.Identifiable;

public enum UserRole implements Identifiable<Integer> {
    Editor(0, "Bearbeiter:in"),
    Publisher(1, "Veröffentlicher:in"),
    Admin(2, "Administrator:in");

    private final Integer key;
    private final String label;

    private UserRole(Integer key, String label) {
        this.key = key;
        this.label = label;
    }

    @Override
    @JsonValue
    public Integer getKey() {
        return key;
    }

    @Override
    public boolean matches(Object other) {
        return key.equals(other);
    }

    public String getLabel() {
        return label;
    }
}
